File: src/main/java/com/zepben/cimbend/database/sqlite/tables/associations/TableUsagePointsEndDevices.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.cimbend.database.sqlite.tables.associations;

import com.zepben.annotations.EverythingIsNonnullByDefault;
import com.zepben.cimbend.database.Column;
import com.zepben.cimbend.database.sqlite.tables.SqliteTable;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static com.zepben.cimbend.database.Column.Nullable.NOT_NULL;

@EverythingIsNonnullByDefault
public class TableUsagePointsEndDevices extends SqliteTable {

    public final Column USAGE_POINT_MRID = new Column(++columnIndex, "usage_point_mrid", "TEXT", NOT_NULL);
    public final Column END_DEVICE_MRID = new Column(++columnIndex, "end_device_mrid", "TEXT", NOT_NULL);

    @Override
    public String name() {
        return "usage_points_end_devices";
    }

    @Override
    public List<List<Column>> uniqueIndexColumns() {
        List<List<Column>> cols = super.uniqueIndexColumns();
        cols.add(Arrays.asList(USAGE_POINT_MRID, END_DEVICE_MRID));
        return cols;
    }

    @Override
    public List<List<Column>> nonUniqueIndexColumns() {
        List<List<Column>> cols = super.nonUniqueIndexColumns();
        cols.add(Collections.singletonList(USAGE_POINT_MRID));
        cols.add(Collections.singletonList(END_DEVICE_MRID));
        return cols;
    }

    @Override
    protected Class<?> getTableClass() {
        return TableUsagePointsEndDevices.class;
    }

    @Override
    protected Object getTableClassInstance() {
        return this;
    }

}

File: src/main/kotlin/com/zepben/evolve/database/sqlite/tables/associations/TableUsagePointsEndDevices.kt
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.evolve.database.sqlite.tables.associations

import com.zepben.evolve.database.sqlite.tables.Column
import com.zepben.evolve.database.sqlite.tables.Column.Nullable.NOT_NULL
import com.zepben.evolve.database.sqlite.tables.SqliteTable

@Suppress("PropertyName")
class TableUsagePointsEndDevices : SqliteTable() {

    val USAGE_POINT_MRID: Column = Column(++columnIndex, "usage_point_mrid", "TEXT", NOT_NULL)
    val END_DEVICE_MRID: Column = Column(++columnIndex, "end_device_mrid", "TEXT", NOT_NULL)

    override fun name(): String {
        return "usage_points_end_devices"
    }

    override fun uniqueIndexColumns(): MutableList<List<Column>> {
        val cols = super.uniqueIndexColumns()

        cols.add(listOf(USAGE_POINT_MRID, END_DEVICE_MRID))

        return cols
    }

    override fun nonUniqueIndexColumns(): MutableList<List<Column>> {
        val cols = super.nonUniqueIndexColumns()

        cols.add(listOf(USAGE_POINT_MRID))
        cols.add(listOf(END_DEVICE_MRID))

        return cols
    }

    override val tableClass: Class<TableUsagePointsEndDevices> = this.javaClass
    override val tableClassInstance: TableUsagePointsEndDevices = this

}
